spring-tx/src/main/java/org/springframework/transaction/interceptor/RuleBasedTransactionAttribute.java

```diff
@@ -147,13 +148,15 @@ public boolean rollbackOn(Throwable ex) {
 			}
 		}
 
-		if (logger.isTraceEnabled()) {
+		if (traceEnabled) {
 			logger.trace("Winning rollback rule is: " + winner);
 		}
 
 		// User superclass behavior (rollback on unchecked) if no rule matches.
 		if (winner == null) {
-			logger.trace("No relevant rollback rule found: applying default rules");
+			if(traceEnabled) {
+				logger.trace("No relevant rollback rule found: applying default rules");
```

Comment:
The above statement won't incur unnecessary perf overhead like in other cases (string concatenation has to be done beforehand) and IMO the logging level checking is not that necessary.

Reply:
Log level check whether it is necessary depends on your logging system.
I think Spring needs this check for compatibility.

Reply:
> The above statement won't incur unnecessary perf overhead like in other cases (string concatenation has to be done beforehand) and IMO the logging level checking is not that necessary.

Log level check whether it is necessary depends on your logging system.
I think Spring needs this check for compatibility.